ext/drx/scatter_gather_aarch64.c

```diff
@@ -419,46 +423,54 @@ expand_scatter_gather(void *drcontext, instrlist_t *bb, instr_t *sg_instr,
     DR_ASSERT_MSG(reg_is_z(sg_info->base_reg) || reg_is_z(sg_info->index_reg),
                   "Base or index must be a Z register");
 
+    DR_ASSERT_MSG(
+        (scalar_src_or_dst != scalar_index_or_base) ||
+            (sg_info->is_load && sg_info->reg_count == 1),
+        "Single register loads are the only instructions where it is safe for the scratch"
+        " registers to alias");
+
     const uint no_of_elements =
         (opnd_size_in_bytes(sg_info->scatter_gather_size) / sg_info->reg_count) /
         opnd_size_in_bytes(sg_info->scalar_value_size);
 
```

Comment:
Add a loop-top comment that here we're trying to zero what we can in the dest reg.

Reply:
Done.